### PR TITLE
add support for prefix parameters

### PR DESCRIPTION
**Pull request by bernt-matthias**
as defined in the standard https://github.com/WorkflowConversion/CTDSchema/blob/5ab34f57cc18cf472aac109e9bdde0e40855dda2/Param_1_7_0.xsd#L100

CTDConverter PR is coming soon and OpenMS may have such parameters at some point (hopefully): https://github.com/OpenMS/OpenMS/pull/4527 .. so we will be prepared

### CODE REVIEW

**Comment by bgruening** [comment]
@timosachsenberg can you please look at this?

File: CTDopts/CTDopts.py [hunk 1]
```diff
@@ -46,12 +46,27 @@ class _OutFile(str):
...
+    pass
+
+
 # module globals for some common operations (python types to CTD-types back and forth)
 TYPE_TO_CTDTYPE = {int: 'int', float: 'double', str: 'string', bool: 'bool',
-                   _InFile: 'input-file', _OutFile: 'output-file'}
-CTDTYPE_TO_TYPE = {'int': int, 'float': float, 'double': float, 'string': str, 'boolean': bool, 'bool': bool,
-                   'input-file': _InFile, 'output-file': _OutFile, int: int, float: float, str: str,
-                   bool: bool, _InFile: _InFile, _OutFile: _OutFile}
+                   _InFile: 'input-file', _OutFile: 'output-file',
```
Comment (jpfeuffer) [inline CTDopts/CTDopts.py:61]: input-prefix missing?

**Comment by jpfeuffer** [comment]
Yes, this is a good idea. For example, SeqAn already supports those prefixes for e.g. indices.
We just did not have a need in OpenMS, yet (although there are one or two candidate tools that would profit from it).

File: CTDopts/CTDopts.py (hunk 1 above)
Comment (bernt-matthias) [reply CTDopts/CTDopts.py:61]: indeed. fixed

**Comment by bernt-matthias** [comment]
> We just did not have a need in OpenMS, yet (although there are one or two candidate tools that would profit from it).

I found it essential for:

- `OpenSwathFileSplitter -outputDirectory`
- `IDRipper -out_path`
- `MzMLSplitter -out`

The others mentioned in https://github.com/OpenMS/OpenMS/issues/4404 seem not so important.

**Comment by bernt-matthias** [comment]
@timosachsenberg would be nice if we could get this merged.

**Comment by timosachsenberg** [comment]
looks good from my side

**Comment by bernt-matthias** [comment]
@timosachsenberg can you also trigger a release? Then I would bump the conda-forge recipe. On the long run an automatic deployment to pypi would be great .. then the conda-forge recipe would auto-bump.

**Pull request merged**